Ports - Ari

aribray opened the pull request:
# Portmanteau Generator
Congratulations! You're submitting your assignment.

## Comprehension Questions

Question 	| Answer
---	      | ---
What went well in your code style, such as indentation, spacing, variable names, readability, etc.? What was lacking? | variable names, spacing, readability. I could have had more effective comments explaining why I was doing what I was doing
How did creating the `is_vowel?` method affect your project? | The method allowed me to find the last and first vowel instances in the words the user gave me.
What was your strategy for getting the correct set of letters from the first word into the portmanteau? | I assigned the index of the last vowel to a variable and used that variable to include/exclude the correct set of letters. 
What was your strategy for validating against inputs under 2 characters? | I used word.length >=2 to get inputs greater than or equal to 2 characters. I guess I technically didn't validate against inputs under 2 characters because I didn't let the user submit anything under 2.
In the next project, what would you change about your process? What would you keep doing? | I did a little bit of pseudocoding, but next time I'm going to read the ENTIRE assignment and requirements first, then go back and break down my strategy and pseudocode a bit before actually trying to type something up.

aribray committed:
Create generator.rb

tildeee commented:
# Portmanteau Generator
## What We're Looking For

Feature  | Feedback
---       | ---
Readable code with consistent indentation | x, gave a small note about indentation
Practices using variables appropriately | x
Practices using conditionals appropriately | x
Practices iteration appropriately | x
Practices using custom methods appropriately | x
Program validates against input under 2 characters | x
Takes in two inputs and creates a portmanteau | x

Hi Ari!

Nice work on this project! I like your solution and your code style. Your code was readable, clean, and logical. I like the structure of your code!

I have a couple of comments and suggestions for code improvement, as well as the discovery of a huge bug in your code! :scream:

That being said, I think your project submission looks great. Great work!

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+# length and vowel conditions are met
+  until word_a.length >= 2 && found_a_vowel == true
+    first_word_array = word_a.split('')
+    if first_word_array.any? {|letter| is_vowel?(letter)}
```
Nice use of the `any?` method!

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+    if first_word_array.any? {|letter| is_vowel?(letter)}
+      found_a_vowel = true
+    else
+      puts "Please enter a word with at least one vowel."
```
Nice work on getting to this optional!

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+    second_word_array = word_b.split('')
+    if second_word_array.any? {|letter| is_vowel?(letter)}
+      found_a_vowel = true
+    end
```
Oh no! Huge bug! What happens if your second word (`word_b`) doesn't have a vowel? To recreate, run your program with the input as first word: `breakfast`, second word: `xx`. It gets stuck in an infinite loop! Why? What's happening? Well, can `found_a_vowel` in this loop ever be `true` in order to exit this loop? I don't think so. I'd like for you to find your bug-- Let me know if you'd like a second pair of eyes to talk this through :)

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+# defining run_generator method
+def run_generator
+
+# the methods I created earlier
```
I would keep your comments on the same level of indentation as the code it aligns with... so this comment should be indented once

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+  last_vowel_index = 0
+  index = 0
+
+# looping through each value in the array to find the last vowel
```
My note about comment indentation being aligned with the other bits of code also applies here

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+
+# the methods I created earlier
+  word_a = get_first_word
+  word_b = get_second_word
```
I love that you made these smaller, helper methods! It helps keep your code clean!

tildeee commented on generator.rb:
```diff
@@ -0,0 +1,118 @@
+  run_generator
+else
+  puts "Thanks for playing!"
+end
```
This actually only allows the user to play twice! If they want to play a third time, the program will quit and not ask again if they want to continue. If you wanted to do that, you'd have to put this code in a loop, not an `if ... else`

aribray committed:
Update generator.rb

Fixed infinite loop in second word vowel check, fixed indentations, and put ability to replay into an until loop that will prompt and run until the answer is no.
